src/test/java/org/codeforamerica/shiba/pages/AbstractPageControllerTest.java

```diff
@@ -73,9 +74,13 @@
 
     @AfterEach
     void cleanup() {
-        applicationData.setSubworkflows(new Subworkflows());
+        applicationData.setId(null);
+        applicationData.setUtmSource(null);
         applicationData.setPagesData(new PagesData());
+        applicationData.setSubworkflows(new Subworkflows());
         applicationData.setIncompleteIterations(new HashMap<>());
         applicationData.setUploadedDocs(new ArrayList<>());
+        applicationData.setFlow(FlowType.UNDETERMINED);
+        applicationData.setSubmitted(false);
```

Comment:
You've probably thought about this, but could we set this as just a blank `ApplicationData`? That way we wouldn't have to remember to clean up more fields if we ever added more attributes to it.

Reply:
Yeah that would be ideal. I haven't figured out how to do it though because we autowire that applicationData into components like the PageController. I'm not sure if we can replace an autowired bean with a brand new one in a way that propagates across all the consumers of that bean.